bittensor/_neuron/text/core_server/run.py

```diff
@@ -366,15 +370,19 @@ def backward_callback(inputs_x:torch.FloatTensor, grads_dy:torch.FloatTensor, sy
         if config.neuron.local_train:
             # --- Training step.
             while end_block >= current_block:
-                if current_block != subtensor.get_current_block():
-                    loss, _ = model( next( dataset ).to(model.device) )
-                    if iteration > 0 : 
-                        losses += loss
-                    else:
-                        losses = loss
-                    iteration += 1
-                    current_block = subtensor.get_current_block()
-                    logger.info(f'local training\titeration: {iteration}\tloss: {loss}')
+                if current_block != subtensor.get_current_block() and axon.priority_threadpool.is_empty:
+                    with mutex:
```

Comment:
Just a clarification required: you're making it so that the entire thread pool must be completely empty to do the backward callback? Would it make a difference if it contained one thread or two threads?

Reply:
Basically we cannot do a local_train and a remote_train at the same time. So the work around is to do a local_train only when the server is free, cause we want to put the remote_train (serving a validator) as the first priority

Reply:
`axon.priority_threadpool` contains all the remote jobs, so when it is empty, it means that the server is free then we can do the local train